### PR TITLE
Issue#11989, #11988, #11990

### PR DESCRIPTION
**Pull request by 007durgesh219**
Before submitting pull request, please check that every commit:
- [x] Has proper Signed-Off-By
- [x] Has commit message which describes it
- [x] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them

### CODE REVIEW

**Comment by nijel** [comment]
Is the patch here different from one in #11994?

File: libraries/URL.php [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+/**
+ * URL/hidden inputs generating.
+ *
+ * @package PhpMyAdmin
+ */
+namespace PMA\libraries;
+
+if (! defined('PHPMYADMIN')) {
+    exit;
+}
```
Comment (nijel) [inline libraries/URL.php]: This is not needed here.

File: test/libraries/PMA_generateCommonUrl_test.php [hunk 2]
```diff
@@ -1,18 +1,17 @@
 <?php
 /* vim: set expandtab sw=4 ts=4 sts=4: */
 /**
- * tests for PMA_URL_getCommon()
+ * tests for URL::getCommon()
```
Comment (nijel) [inline test/libraries/PMA_generateCommonUrl_test.php]: I think this file should be renamed to test/classes/URL.php

File: test/libraries/PMA_js_escape_test.php [hunk 3]
```diff
@@ -9,7 +9,7 @@
 /*
  * Include to test.
  */
-require_once 'libraries/js_escape.lib.php';
+use PMA\libraries\Sanitize;
```
Comment (nijel) [inline test/libraries/PMA_js_escape_test.php]: I think this file should be renamed to test/classes/Sanitize.php

File: test/libraries/PMA_sanitize_test.php [hunk 4]
```diff
@@ -1,20 +1,17 @@
 <?php
 /* vim: set expandtab sw=4 ts=4 sts=4: */
 /**
- * tests for PMA_sanitize()
+ * tests for Sanitize::sanitize()
```
Comment (nijel) [inline test/libraries/PMA_sanitize_test.php]: I think this file should be renamed to test/classes/Sanitize.php (and merged with js based tests)

**Comment by nijel** [comment]
This looks good, but I've some minor comments, see above.

**Comment by 007durgesh219** [comment]
Thanks @nijel  I will update PR shortly

**Comment by nijel** [comment]
PS: There seems to be also merge conflict meanwhile, it would be great if you could rebase it on current master.

**Comment by 007durgesh219** [comment]
@nijel I've done required changes

**Comment by nijel** [comment]
What you've not yet done is suggested file renaming ;-)

File: test/classes/URL.php [hunk 5]
```diff
@@ -1,18 +1,17 @@
 <?php
```
Comment (007durgesh219) [inline test/classes/URL.php]: changed test/libraries/PMA_generateCommonUrl_test.php → test/classes/URL.php

File: test/classes/Sanitize.php [hunk 6]
```diff
@@ -1,20 +1,17 @@
 <?php
```
Comment (007durgesh219) [inline test/classes/Sanitize.php]: this one also  test/libraries/PMA_sanitize_test.php → test/classes/Sanitize.php

**Comment by 007durgesh219** [comment]
@nijel I think you suggested only those two. also, I've merged PMA_js_escape_test to Sanitize.php .  Is there something else I am missing??

**Comment by 007durgesh219** [comment]
This seems a bug in the github. If we change only file name then it won't show it as outdated diff

**Comment by nijel** [comment]
Okay, I somehow trusted GitHub in this, sorry. Merging the changes right now...

**Pull request merged**